src/res_core.ml

```diff
@@ -1374,11 +1396,10 @@ and parseConstrainedPatternRegion p =
  *	 | field , _
  *	 | field , _,
  *)
-and parseRecordPatternField p =
-  let label = parseValuePath p in
+and parseRecordPatternField ~(label: Longident.t Location.loc) p =
```

Comment:
We do the label thing in the caller, which has more context on what the token is.